bigquery/google/cloud/bigquery/table.py

```diff
@@ -1334,8 +1339,26 @@ def _to_dataframe_tabledata_list(self, dtypes):
         """Use (slower, but free) tabledata.list to construct a DataFrame."""
         column_names = [field.name for field in self.schema]
         frames = []
+
+        # report progress if tqdm installed
+        pbar = None
+        if tqdm is not None:
+            try:
+                pbar = tqdm.tqdm(
+                    desc="Downloading", total=self.total_rows, unit="rows"
+                )
+            except (KeyError, TypeError):
+                # tqdm error
+                pass
+
         for page in iter(self.pages):
             frames.append(self._to_dataframe_dtypes(page, column_names, dtypes))
+
+            if pbar is not None:
+                pbar.total = pbar.total or self.total_rows
```

Comment:
We should add a comment here explaining why we are setting the total (again).

```
	                # In some cases, the number of total rows is not populated
	                # until the first page of rows is fetched. Update the
	                # progress bar's total to keep an accurate count.
	                pbar.total = pbar.total or self.total_rows
```

Reply:
Will do!